src/accelerate/test_utils/scripts/external_deps/test_metrics.py

```diff
@@ -149,6 +149,40 @@ def test_mrpc(dispatch_batches: bool = False, split_batches: bool = False):
         ), f"Baseline and Distributed are not the same for key {key}:\n\tBaseline: {baseline[key]}\n\tDistributed: {distributed[key]}\n"
 
 
+def test_gather_for_metrics_with_non_tensor_objects_iterable_dataset():
```

Comment:
I think the test is not being run at all, right? Similar to `test_gather_for_metrics_with_iterable_dataset`, it needs to be explicitly called in the `main` function, because the tests in this file are _not_ collected by pytest but rather called directly.